marlowe-playground-client/src/JavascriptEditor/View.purs

```diff
@@ -86,12 +98,29 @@ jsEditor state = slot _jsEditorSlot unit mkEditor unit (Just <<< HandleEditorMes
 
 compileButton :: forall p. State -> HTML p Action
 compileButton state =
-  button [ onClick $ const $ Just Compile ]
-    [ text (if state ^. _compilationResult <<< to isLoading then "Compiling..." else "Compile") ]
+  button
+    [ onClick $ const $ Just Compile
+    , enabled enabled'
+    , classes classes'
+    ]
+    [ text buttonText ]
   where
-  isLoading JS.Compiling = true
+  buttonText = case view _compilationResult state of
+    JS.Compiling -> "Compiling..."
+    JS.CompiledSuccessfully _ -> "Compiled"
+    JS.CompilationError _ -> "Compiled"
```

Comment:
are we sure? maybe it should be "Compile" if there is an error?

Reply:
It is "Compiled" in red.... which technically is true. It doesn't allow you to send to the simulator (nor compile again until you make some modification)